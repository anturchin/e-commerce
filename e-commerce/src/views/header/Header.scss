.header {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 1.5rem 0 1.5rem 0;
    box-shadow: 0 15px 35px rgb(0, 0, 0);
    background: #222;
    color: #0f0;

    position: relative;
    min-height: 7%;
}

@media (max-width: 485px) {
    .header {
        flex-direction: column;
        padding: 1rem 4rem;
        align-items: flex-end;
    }
}
