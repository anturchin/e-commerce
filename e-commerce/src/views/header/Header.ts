import { View } from '../View';
import { NavList } from './navList/NavList';
import { SvgBag } from './imgBag/ImgBag';
import { Button } from '../button/Button';

import './Header.scss';

export class Header extends View {
    private navList: NavList | null = null;

    private imgBag: HTMLElement | null = null;

    private toggleButton: Button | null = null;

    constructor() {
        super({ tag: 'header', classNames: ['header'] });
        this.setupHeaderContent();
        this.setupToggleBtn();
        this.setupResizeListener();
        this.updateNavList();
    }

    public getNavList(): NavList | null {
        return this.navList;
    }

    public getImgBag(): HTMLElement | null {
        return this.imgBag;
    }

    private setupHeaderContent(): void {
        this.navList = new NavList();
        this.imgBag = new SvgBag().getElement();
        this.viewHtmlElement.addInnerElement(this.navList.getElement());
        this.viewHtmlElement.addInnerElement(this.imgBag);
    }

    private setupToggleBtn(): void {
        this.toggleButton = new Button({
            label: 'â˜°',
            onClick: () => this.toggleNavList(),
        });
        this.viewHtmlElement.addInnerElement(this.toggleButton.getElement());
        this.updateBtnVisible();
    }

    private setupResizeListener(): void {
        window.addEventListener('resize', () => {
            this.updateBtnVisible();
            this.updateNavList();
        });
    }

    private updateBtnVisible(): void {
        const buttonElement = this.toggleButton?.getElement();
        if (buttonElement) {
            if (window.innerWidth <= 485) {
                buttonElement.style.display = 'block';
            } else {
                buttonElement.style.display = 'none';
            }
        }
    }

    private updateNavList(): void {
        const navListElement = this.navList?.getElement();
        if (navListElement) {
            navListElement.style.display = window.innerWidth <= 485 ? 'none' : 'flex';
        }
    }

    private toggleNavList(): void {
        if (this.navList) {
            const navElement = this.navList.getElement();
            navElement.style.display = navElement.style.display === 'none' ? 'flex' : 'none';
        }
    }
}
