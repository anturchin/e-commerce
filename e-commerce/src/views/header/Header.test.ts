import { describe, expect, test } from '@jest/globals';
import { Header } from './Header';

describe('Header', () => {
    test('should match', () => {
        const header = new Header();
        const renderedHeader = header.getElement();
        expect(renderedHeader.outerHTML).toMatchSnapshot();
    });
});
