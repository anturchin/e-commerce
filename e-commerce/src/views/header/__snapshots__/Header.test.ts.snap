// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match 1`] = `"<header class="header"><ul class="nav-list hidden" style="display: flex;"><li class="nav-list_item" data-nav-item="Home">Home</li><li class="nav-list_item" data-nav-item="Category">Category</li><li class="nav-list_item" data-nav-item="Sign in">Sign in</li><li class="nav-list_item" data-nav-item="Sign up">Sign up</li><li class="nav-list_item" data-nav-item="Profile">Profile</li><li class="nav-list_item" data-nav-item="About">About</li><li class="nav-list_item" data-nav-item="Logout">Logout</li></ul><div class="nav-list__item-container"><img src="svg-mock" data-nav-item="Bag" class="nav-list__item-bag"><h5 id="cart_length"></h5></div><button class="custom-button" style="display: none;">â˜°</button></header>"`;
